source/mbed_trace.c

```diff
@@ -162,9 +162,6 @@ int mbed_trace_init(void)
     m_trace.suffix_f = 0;
     m_trace.printf = mbed_trace_default_print;
     m_trace.cmd_printf = 0;
-    m_trace.mutex_wait_f = 0;
-    m_trace.mutex_release_f = 0;
-    m_trace.mutex_lock_count = 0;
```

Comment:
Ok, does not belong to this PR, but I would use here something like "NULL-object" pattern to avoid `if (m_trace.mutex_wait_f) {...}` everywhere